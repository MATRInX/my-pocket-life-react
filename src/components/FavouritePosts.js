import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectFavouritePosts from '../selectors/favouritePosts';

export const FavouritePosts = ({ posts } = props) => {
  return (
    <div>
      <p>FavouritePosts content</p>
      <ul>
        {
          posts.map((post, index) => (
            <Link key={index} to={`/post${post.url.substr(post.url.lastIndexOf('/'))}`}>
              <h5>{post.title}</h5>
              <img src={post.image} />
            </Link>
          ))
        }
      </ul>
    </div>
  )
};

const mapStateToProps = ({ posts } = state) => ({
  posts: selectFavouritePosts(posts)
})

export default connect(mapStateToProps)(FavouritePosts);