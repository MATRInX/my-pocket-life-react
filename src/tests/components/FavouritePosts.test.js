import React from 'react';
import { shallow } from 'enzyme';
import FavouritePosts from '../../components/FavouritePosts';

test('should correctly render FavouritePosts widget on sidebar', () => {
  const wrapper = shallow(<FavouritePosts />);
  expect(wrapper).toMatchSnapshot();
});