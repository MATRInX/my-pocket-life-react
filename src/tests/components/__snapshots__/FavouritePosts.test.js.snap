// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render FavouritePosts widget on sidebar 1`] = `
<div>
  FavouritePosts content
</div>
`;
